File: Basic/16.Find all pairs in an array whose sum is equal to a given number/Solution.cpp
#include <iostream>
#include <vector>
using namespace std;

void findPairs(vector<int>& arr, int target) {
    bool found = false;
    cout << "Pairs with sum " << target << ":\n";
    for (int i = 0; i < arr.size() - 1; i++) {
        for (int j = i + 1; j < arr.size(); j++) {
            if (arr[i] + arr[j] == target) {
                cout << "(" << arr[i] << ", " << arr[j] << ")\n";
                found = true;
            }
        }
    }
    if (!found) {
        cout << "No pairs found.\n";
    }
}

int main() {
    int n, target;
    cout << "Enter the size of the array: ";
    cin >> n;

    vector<int> arr(n);
    cout << "Enter " << n << " elements of the array: ";
    for (int i = 0; i < n; i++) {
        cin >> arr[i];
    }

    cout << "Enter the target sum: ";
    cin >> target;

    findPairs(arr, target);

    return 0;
}

File: Basic/16.Find all pairs in an array whose sum is equal to a given number/Solution.py
def find_pairs(arr, target):
    found = False
    print(f"Pairs with sum {target}:")
    for i in range(len(arr) - 1):
        for j in range(i + 1, len(arr)):
            if arr[i] + arr[j] == target:
                print(f"({arr[i]}, {arr[j]})")
                found = True
    if not found:
        print("No pairs found.")

if __name__ == "__main__":
    n = int(input("Enter the size of the array: "))
    arr = list(map(int, input(f"Enter {n} elements of the array: ").split()))

    target = int(input("Enter the target sum: "))
    find_pairs(arr, target)
